src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs

```diff
@@ -1272,6 +1272,37 @@ public static class ILoggerExtensions
             LogAzureSearchDependency(logger, searchServiceName, operationName, isSuccessful, measurement.StartTime, measurement.Elapsed, context);
         }
 
+        /// <summary>
+        /// Logs an Azure Search Dependency.
+        /// </summary>
+        /// <param name="logger">The logger to track the telemetry.</param>
+        /// <param name="searchServiceName">The name of the Azure Search service.</param>
+        /// <param name="operationName">The name of the operation to execute on the Azure Search service.</param>
+        /// <param name="isSuccessful">The indication whether or not the operation was successful.</param>
+        /// <param name="measurement">The measuring the latency to call the dependency.</param>
+        /// <param name="dependencyId">The ID of the dependency to link as parent ID.</param>
+        /// <param name="context">The context that provides more insights on the dependency that was measured.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="logger"/> or <paramref name="measurement"/> is <c>null</c>.</exception>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="searchServiceName"/> or <paramref name="operationName"/> is blank.</exception>
+        public static void LogAzureSearchDependency(
+            this ILogger logger,
+            string searchServiceName,
+            string operationName,
+            bool isSuccessful,
+            DurationMeasurement measurement,
+            string dependencyId,
```

Comment:
So `dependencyId` is used to link this dependency to a certain parent ?
I assume that the name of this parameter is consistent with the name of related parameters in other methods, but I wonder if we can improve the name of the parameter to make this more clear ?
If this parameter is used to pass in the Id of the parent, so that we can link it to the parent, shouldn't we call it `parentId` ?

Reply:
It's actually the way around, the ID is to link a request to the dependency. So, we have 'outgoing dependencies' and 'incoming requests'. This ID will be the parent ID of the request tracking. So, when tracking requests, it's set as the parent ID. In here, with dependencies, it's only to identify the dependency.

Reply:
I don't understand.  Maybe we should discuss this offline.

Reply:
Yeah, sure. It's something that we already discussed twice in separate meetings, 😅 . I hope when we get through with the service-to-service correlation, we can create some kind of diagram to make this more clear.

Reply:
Discussed offline, we can keep on doing this adding of dependency ID's to the dependency tracking.

Reply:
Just to be sure: we also have an overload which doesn't have the dependencyId; that overload generates an id himself ?
If I look at the code, that is not the case.

Reply:
> Just to be sure: we also have an overload which doesn't have the dependencyId; that overload generates an id himself ?

No, we have an overload w/o passing a dependency ID but we can't generate this as it also has to be passed-along with the HTTP request, the Azure Service Bus message, .... or other 'request message' towards a dependent system.

Reply:
Yes.  I get it.